metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py

```diff
@@ -134,15 +151,47 @@ def __init__(self, config: IcebergSourceConfig, ctx: PipelineContext) -> None:
         super().__init__(config, ctx)
         self.report: IcebergSourceReport = IcebergSourceReport()
         self.config: IcebergSourceConfig = config
+        self.ctx: PipelineContext = ctx
 
     @classmethod
     def create(cls, config_dict: Dict, ctx: PipelineContext) -> "IcebergSource":
         config = IcebergSourceConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
     def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
+        auto_lowercase_dataset_urns: Optional[MetadataWorkUnitProcessor] = None
```

Comment:
may you add some comment explaining why this reimplementation of the `get_workunit_processors` is needed as well as why the system metadata stamp must be decoupled from the workunit processors?

Reply:
Done